crates/context/config/src/client/env/proxy/query/context_storage_entries.rs

```diff
@@ -167,13 +166,14 @@ impl Method<Evm> for ContextStorageEntriesRequest {
         Ok(SolValue::abi_encode(&(offset, limit)))
     }
     fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
-        let decoded: Vec<B256> = SolValue::abi_decode(&response, false)?;
+        let decoded: Vec<alloy::primitives::Bytes> = SolValue::abi_decode(&response, false)?;
+        let decoded: Vec<Vec<u8>> = decoded.into_iter().map(|b| b.to_vec()).collect();
 
         let context_storage_entries = decoded
             .chunks(2)
             .map(|chunk| ContextStorageEntry {
-                key: chunk[0].to_vec(),
-                value: chunk[1].to_vec(),
+                key: chunk[0].clone(),
+                value: chunk[1].clone(),
             })
             .collect();
```

Comment:
```
let decoded: Vec<Vec<u8>> = decoded.into_iter().map(|b| b.to_vec()).collect();
```

^^^ this will reallocate for every byte

```
key: chunk[0].clone(),
value: chunk[1].clone(),
```

^^^ this will reallocate again for those bytes

[`Bytes: Into<Vec<u8>>`](https://docs.rs/alloy/latest/alloy/primitives/struct.Bytes.html#impl-From%3CBytes%3E-for-Vec%3Cu8%3E), so instead we can write this:

```suggestion
        let mut decoded = decoded.into_iter();

        let mut entries = Vec::with_capacity(decoded.len() / 2);

        while let Some(key) = decoded.next() {
            let value = decoded.next().ok_or_eyre("missing value for storage entry")?;
            
            entries.push(ContextStorageEntry {
                key: key.into(),
                value: key.into(),
            ));
        }
```